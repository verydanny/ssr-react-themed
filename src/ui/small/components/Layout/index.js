export Layout from './components/Layout'
export Header from './components/Header'
export HomeHeader from './components/HomeHeader'
