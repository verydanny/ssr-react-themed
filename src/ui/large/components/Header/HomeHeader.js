import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import themed from 'react-themed-too'
import { Icon } from 'ui/shared/components/Icon'
import rentComLogo from 'ui/shared/components/Icon/svgs/rentComLogo.svg'
import { MENU_LINKS } from 'ui/large/components/Header/const'
import withUserData from 'ui/large/components/Header/withUserData'
import HeaderBadge from './components/HeaderBadge'

const HOME_MENU_LINKS = MENU_LINKS.filter(item => item.name !== 'Search')

@themed(/^Header|HomeHeader/)
class HomeHeader extends PureComponent {
  static propTypes = {
    theme: PropTypes.object,
    favorites: PropTypes.object,
  }

  static defaultProps = {
    theme: {},
  }

  render() {
    const { theme, favorites } = this.props
    const numFavorites = Object.keys(favorites).length

    return (
      <header className={theme.HomeHeader} data-tag_section="header">
        <div className={theme.HomeHeader_Content}>
          <a className={theme.Header_rentComLogoLink} href="/">
            <Icon svgs={{ rentComLogo }} className={theme.Header_rentComLogo} data-tid="rent-logo" />
          </a>
          <div>
            {HOME_MENU_LINKS.map((menu, index) => (
              <a
                key={`menu-${index}`}
                className={theme.Header_Menulink}
                href={menu.link}
                data-tag_item={menu.tag}
              >
                {menu.name}
                {menu.hasBadge && !!numFavorites ? (
                  <HeaderBadge numFavorites={numFavorites} />
                ) : null}
              </a>
            ))}
          </div>
          {/*
            // TODO: restore Sign In logic when stories are ready
            <div className={theme.Button_Signin} data-tag_item="sign_in">
              Sign In
            </div>
            */}
        </div>
      </header>
    )
  }
}

export default withUserData(HomeHeader)
