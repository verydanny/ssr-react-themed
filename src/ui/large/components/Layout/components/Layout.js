import React from 'react'
import PropTypes from 'prop-types'
import ModalStack from 'ui/shared/components/ModalStack'
import { modalDefinitions } from 'ui/large/config/modalConfig'
import { Header as DefaultHeader } from 'ui/large/components/Header'
import { themed } from 'react-themed-too'

const Layout = props => (
  <div className={props.theme.Layout_Container}>
    <ModalStack modalDefinitions={modalDefinitions} />
    {props.HeaderComponent && <props.HeaderComponent />}
    <section className={props.theme.Layout_Content}>
      { props.children }
    </section>
  </div>
)

Layout.propTypes = {
  children: PropTypes.node,
  HeaderComponent: PropTypes.any,
  theme: PropTypes.object,
}

Layout.defaultProps = {
  theme: {},
  HeaderComponent: DefaultHeader,
}

export default themed(/^Layout/)(Layout)
