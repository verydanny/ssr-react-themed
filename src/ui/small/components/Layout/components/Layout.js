import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'react-themed'
import classNames from 'classnames'
import theme from 'ui/small/themes/SmallTheme'
import ModalStack from 'ui/shared/components/ModalStack'
import { modalDefinitions } from 'ui/small/config/modalConfig'
import Header from './Header'

const optionalShape = PropTypes.oneOfType([
  PropTypes.bool,
  PropTypes.node,
])

class Layout extends Component {

  static propTypes = {
    className: PropTypes.string,
    children: PropTypes.node.isRequired,
    headerComponent: optionalShape,
    footerComponent: optionalShape,
    layout: PropTypes.object,
  }

  static defaultProps = {
    headerComponent: <Header />,
    layout: {},
  }

  render() {
    const {
      headerComponent,
      footerComponent,
      children,
      className,
      layout,
    } = this.props

    // Theme is provided for everything under the Layout component.
    // However also refer to ui/small/components/ActiveView.js
    // because it manually applies theme in some cases.

    return (
      <ThemeProvider theme={theme}>
        <div className={classNames(className, layout.Layout_Container)}>
          <ModalStack modalDefinitions={modalDefinitions} />
          {headerComponent &&
            <header className={layout.Layout_Header}>
              {
                headerComponent
              }
            </header>
          }
          <section className={layout.Layout_Content}>
            {children}
          </section>
          {footerComponent &&
            <footer className={layout.Layout_Footer}>
              {
                footerComponent
              }
            </footer>
          }
        </div>
      </ThemeProvider>
    )
  }
}

export default Layout
