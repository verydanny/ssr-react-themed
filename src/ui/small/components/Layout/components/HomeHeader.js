import React, { Component } from 'react'
import autobind from 'autobind-decorator'
import PropTypes from 'prop-types'
import themed from 'react-themed-too'
import Menu from 'ui/small/components/Menu'
import { Icon } from 'ui/shared/components/Icon'
import rentComLogo from 'ui/shared/components/Icon/svgs/rentComLogo.svg'
import menu from 'ui/shared/components/Icon/svgs/menu.svg'

@themed(/^HomeHeader/)
class HomeHeader extends Component {
  static propTypes = {
    theme: PropTypes.object,
  }

  static defaultProps = {
    theme: {},
  }

  constructor(props) {
    super(props)

    this.state = {
      displayMenu: false,
    }
  }

  @autobind
  toggleMenu() {
    this.setState({
      displayMenu: !this.state.displayMenu,
    })
  }

  render() {
    const { theme } = this.props

    return (
      <div className={theme.HomeHeader} data-tag_section="header">
        <Icon
          svgs={{ rentComLogo }}
          className={theme.HomeHeader_Logo}
          data-tid="rent-logo"
        />
        <Icon
          svgs={{ menu }}
          className={theme.HomeHeader_Hamburger}
          data-tid="menu"
          onClick={this.toggleMenu}
          data-tag_item="open_close_menu"
          data-tag_section="menu"
        />
        <Menu
          className={theme.HomeHeader_MenuTransform}
          display={this.state.displayMenu}
          toggleMenu={this.toggleMenu}
        />
      </div>
    )
  }
}

export default HomeHeader
