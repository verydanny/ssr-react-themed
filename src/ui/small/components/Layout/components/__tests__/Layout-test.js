import React from 'react'
import { shallow } from 'enzyme'
import Layout from '../Layout'
import HomeHeader from '../HomeHeader'

describe('small/Layout', () => {
  const props = {
    headerComponent: null,
    footerComponent: null,
  }

  const wrapper = shallow(<Layout {...props}><span id="foo" /></Layout>)

  it('renders children', () => {
    const expected = 1
    const actual = wrapper.find('#foo').length
    expect(actual).toEqual(expected)
  })
})

describe('small/HomeHeader', () => {
  const wrapper = shallow(<HomeHeader />)

  it('renders', () => {
    const actual = wrapper.html()
    expect(actual).toMatchSnapshot()
  })
})
