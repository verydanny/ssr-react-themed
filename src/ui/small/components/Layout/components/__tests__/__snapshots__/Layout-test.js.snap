// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`small/HomeHeader renders 1`] = `

<div data-tag_section="header">
  <div class="undefined undefined"
       data-tid="mobile-menu"
       role="presentation"
  >
    <div>
      <a href="/"
         data-tid="home-link"
         data-tag_item="home"
      >
        Home
      </a>
      <a href="/account/myrent/"
         data-tag_item="my_rent"
      >
        My Rent
      </a>
      <a href="/moving-center/"
         data-tag_item="moving_center"
      >
        Moving Center
      </a>
      <a href="/manage"
         data-tag_item="list_a_property"
      >
        List a property
      </a>
      <a href="/blog/"
         data-tag_item="blog"
      >
        Blog
      </a>
    </div>
  </div>
</div>

`;
